### ISSUE TITLE
Integrated console does not write output on Windows Insider build 16193

### ISSUE THREAD

**Issue by yhvicey** [PowerShell/vscode-powershell#742]
### System Details

- Operating system name and version: Microsoft Windows [Version 10.0.16193.1001]
- VS Code version: 1.12.1
- PowerShell extension version: 1.0.0
- Output from `$PSVersionTable`: 

![image](https://cloud.githubusercontent.com/assets/8000820/26030714/fe1c5672-388c-11e7-80af-21a450639f72.png)

### Issue Description

VSCode cannot launch integrated powershell and stuck at showing "starting powershell..." like this:

![image](https://cloud.githubusercontent.com/assets/8000820/26030716/2b44aae6-388d-11e7-9053-5a39b1e6aab0.png)

### Attached Logs

[log.zip](https://github.com/PowerShell/vscode-powershell/files/999056/log.zip)

**Comment by jaapbrasser** [comment]
I have the same issue on the 10.0.16193.1001 build and also on the previous Insider Fast Ring build. Does anyone know of a workaround for this issue?

**Comment by daviwil** [comment]
I was able to repro the issue on Friday, looks like Windows broke the console APIs that I use to read input.  I was trying to install a newer canary build to test with but it kept rolling back.  I'll try again today to see if the latest builds fix the issue.

**Comment by daviwil** [comment]
Just tried this with Insiders build 16199 and I'm still seeing the problem.  I'll try to debug the issue and see where it's coming from.

/cc @bgelens

**Comment by jaapbrasser** [comment]
Thanks for following up on this @daviwil, I appreciate that!

**Comment by mud5150** [comment]
@jaapbrasser How are you killing the integrated console? Mine comes back whenever I try to debug. Having no luck trying to work around this issue.

**Comment by jaapbrasser** [comment]
![image](https://cloud.githubusercontent.com/assets/12744735/26114718/5285c130-3a5e-11e7-9a3d-50473342b7ed.png)

This is what I do, click this button and then hit **<CTRL>** + **`**

![image](https://cloud.githubusercontent.com/assets/12744735/26114824/9fb7d038-3a5e-11e7-9b11-ec261ff5c4b6.png)

I do end up with this error message, so it is not a full workaround as this blocks me from seeing my open files.

**Comment by jdhitsolutions** [comment]
I still see this issue with VSCode 1.12.2

**Comment by daviwil** [comment]
Updates to VS Code won't fix this unfortunately, it's an issue that manifests in PowerShell Editor Services.

**Comment by daviwil** [comment]
## Workaround

OK folks, I've got a workaround for this issue.  Open up your VS Code profile at one of the following paths (create the file if it doesn't already exist):

**Windows PowerShell 5.1**
```
$HOME\Documents\WindowsPowerShell\Microsoft.VSCode_profile.ps1
```

**PowerShell Core 6.0**
```
$HOME\Documents\PowerShell\Microsoft.VSCode_profile.ps1
```

Insert this line at the top of the file:

```powershell
[System.Console]::OutputEncoding = [System.Text.Encoding]::ASCII
```

Now restart VS Code.  When the Integrated Console appears, you should start to see output again.

## Explanation

Something happened in the latest Windows Insider builds to cause output in the Integrated Console to not be written inside of VS Code's integrated terminal.  My guess is that there's some specific set of factors that contribute to this issue since it doesn't happen when you set powershell.exe to use UTF-8 output encoding inside of VS Code.  I'll continue investigating this issue to try and find a real solution, but the workaround should suffice for now.

**Comment by mud5150** [comment]
@daviwil Thanks man. That got me going.

**Comment by daviwil** [comment]
Thanks for confirming, glad it helped!

**Comment by brwilkinson** [comment]
Thanks worked for me. . .  well after I enabled the profile again.

"powershell.enableProfileLoading": true

**Comment by jaapbrasser** [comment]
Also resolved it for me, you are a legend @daviwil, thanks for looking into this!

**Comment by Graham-Beer** [comment]
Thank you @daviwil, workaround worked a treat.

**Comment by DexterPOSH** [comment]
Aha! It works, now I can start moving away from the ConEmu 😄 .

**Comment by MaximoTrinidad** [comment]
@daviwil !  Thanks for the workaround. It work on both VS Code April release and in the latest VS Code Insider.

:)

**Comment by michelderooij** [comment]
Bit late to the party, but thanks for the workaround (y)

**Comment by ZodmanPerth** [comment]
Workaround doesn't help me at work because the script is blocked by group policy.
Will this be fixed in the next release?

**Comment by jaapbrasser** [comment]
Even without administrative right you can change your ExecutionPolicy on the system, if you run the following you should be fine:

    Set-ExecutionPolicy -ExecutionPolicy Bypass -Scope CurrentUser

**Comment by daviwil** [comment]
@ZodmanPerth not much I can personally do about it right now, this is an issue caused by changes to Windows' console implementation in the latest Insider builds.  I'm still waiting for a fix on their side.

**Comment by daviwil** [comment]
Actually, I could possibly do a temporary fix inside of the language server to change the thing I'm doing that's triggering Windows console's bug, let me look into it for this week's release.

**Comment by bgelens** [comment]
Wanted to provide an update on this from the end-user exp:

without workaround it still doesn't work on Windows build ```16215``` with vs code ```1.13.0``` and PS Ext ```1.3.1```

An update though which didn't work before. I am now able to get a descent working prompt again by first setting ASCII mode and after setting Unicode mode.
![2017-06-09_20-50-05](https://user-images.githubusercontent.com/10071039/27009807-c2acd30c-4e97-11e7-85bb-47f8ef91a4b9.gif)

No more question marks is worth a lot for me!

Note that I cannot go to Unicode directly for some reason.

**Comment by daviwil** [comment]
Awesome, thanks Ben!  So is the output fixed for the rest of the session after that?  I might be able to just make that the fix internally.

**Comment by bgelens** [comment]
Yes, the rest of the session look normal now

**Comment by bgelens** [comment]
@daviwil I got some Chinese chars in the mix when doing stuff with git now.. maybe better to sit out Windows Insider builds until fixed!

**Comment by daviwil** [comment]
Thanks for the update!  I agree, seems like the console components are undergoing a lot of change at the moment so it might be best to avoid the latest Windows Insider builds.

**Comment by MaximoTrinidad** [comment]
Just an FYI

Still happening with PowerShell extension 1.3.2.

![vscodeposhinter_2017-06-13_10-05-22](https://user-images.githubusercontent.com/19892435/27089813-7c8f5a14-5029-11e7-8c01-13e6cedcba19.jpg)

**Comment by Jaykul** [comment]
Still a problem on 1.4.1 on Build 16237

![image](https://user-images.githubusercontent.com/192942/28235828-528c6882-68e4-11e7-9e1c-a852e5be7e14.png)

**Comment by Jaykul** [comment]
But it's fixed in 16241 👍 
![image](https://user-images.githubusercontent.com/192942/28236387-1d26f8f6-68f3-11e7-8f3a-2e808aeba45a.png)

**Comment by bgelens** [comment]
Confirm fix 👍

**Comment by daviwil** [comment]
That's excellent news, thanks a lot for letting me know!
